src/classes/RD2_OpportunityEvaluationService_TEST.cls

```diff
@@ -238,17 +245,212 @@ private class RD2_OpportunityEvaluationService_TEST {
+        };
+
+        RD2_Settings_TEST.setUpConfiguration(rdSettingMap);
+        
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact()).build();
+        
+        Test.startTest();
+        insert rd;
+        Test.stopTest();
+
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+
+        System.assertEquals(0, opps.size(),
+            'No installment opp should be created on insert of RD when the auto Creation setting is ' + OPTION_NAME_DISABLE_FIRST);
+    }
+
+    /**
+    * @description When Installment Auto Creation Setting is set to Disable First Installment, 
+    * the installment should be created on RD when the nightly batch job run
+    */
+    @IsTest
+    private static void shouldCreateOppDuringBatchRunWhenDisableFirstInstallment() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<String, Object> rdSettingMap = new Map<String, Object> {
+            'InstallmentOppAutoCreateOption__c' => OPTION_NAME_DISABLE_FIRST
+        };
+
+        RD2_Settings_TEST.setUpConfiguration(rdSettingMap);
+        
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact()).build();
+        insert rd;
+
+        Test.startTest();
+        Database.executeBatch(new RD2_OpportunityEvaluation_BATCH());
+        Test.stopTest();
+
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+
+        System.assertEquals(1, opps.size(),
+            'The next installment should be created during the batch run when Installment Auto Creation setting is ' + OPTION_NAME_DISABLE_FIRST);
+    }
+
+    /**
+    * @description When Installment Auto Creation Setting is set to Disable All Installments, 
+    * no installment should be created on inserted of RD and during nightly batch job
+    */
+    @IsTest
+    private static void shouldNotCreateOppOnInsertWhenDiableAllInstallment() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<String, Object> rdSettingMap = new Map<String, Object> {
+            'InstallmentOppAutoCreateOption__c' => OPTION_NAME_DISABLE_ALL
+        };
+
+        RD2_Settings_TEST.setUpConfiguration(rdSettingMap);
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact()).build();
+
+        Test.startTest();
+        insert rd;
+        Test.stopTest();
+        
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+
+        System.assertEquals(0, opps.size(),
+            'The installment opportunity should not be created when the Auto Creation Setting is' + OPTION_NAME_DISABLE_ALL);
+    }
+
+    /**
+    * @description When Installment Auto Creation Setting is set to Disable All Installments, 
+    * no installment should be created on inserted of RD and during nightly batch job
+    */
+    @IsTest
+    private static void shouldNotCreateOppDuringBatchRunWhenDiableAllInstallment() {
```

Comment:
Let's move this test verifying the batch `RD2_OpportunityEvaluation_BATCH` into `RD2_OpportunityEvaluation_TEST`.